### DISCUSSION TITLE
Unable to connect to wifi CLR_E_INVALID_OPERATION

### DISCUSSION THREAD

**Discussion by jasonkaisersmith** [nanoframework/Home#1280] [Q&A] [answered]
### Library/API/IoT binding

nanoFramework.Networking

### Visual Studio version

VS2022

### .NET nanoFramework extension version

2022.3.0.41

### Target name(s)

ESP_WROOM_32

### Firmware version

1.8.1.148

### Description

Seems to be potentially the same issue as reported here:
https://github.com/nanoframework/Home/issues/1166

But that was closed.

Output

    ++++ Exception System.InvalidOperationException - CLR_E_INVALID_OPERATION (1) ++++
    ++++ Message: 
    ++++ System.Device.Wifi.WifiAdapter::NativeScanAsync [IP: 0000] ++++
    ++++ System.Device.Wifi.WifiAdapter::ScanAsync [IP: 000d] ++++
    ++++ nanoFramework.Networking.WifiNetworkHelper::ScanAndConnectDhcp [IP: 000d] ++++
    ++++ WebRequest.Program::Main [IP: 0021] ++++

### Sample project or code

            bool success;
            CancellationTokenSource cs = new(60000);

            success = WifiNetworkHelper
                .ConnectDhcp(MySsid, MyPassword, requiresDateTime: true, token: cs.Token);

**Comment by alberk8** [comment]
May I know which firmware was installed on the ESP32?

**Comment by jasonkaisersmith** [comment]
Firmware version
ESP32 (ESP32-D0WD-V3 (revision 3))
1.8.1.148

**Comment by alberk8** [comment, marked as answer]
> Firmware version ESP32 (ESP32-D0WD-V3 (revision 3)) 1.8.1.148

try install firmware with `nanoff --target ESP32_REV3 --update --serialport COM6`

**Comment by jasonkaisersmith** [comment]
Now working.  Thanks.

**Comment by Ellerbach** [comment]
As it's working, closing this discussion.
